components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/RESTServiceUtils.java

```diff
@@ -0,0 +1,478 @@
+	 * @param wadlName      wadl name.
+	 * @param version       wadl version.
+	 * @param wadlPath      wadl path.
+	 * @return              REST Service element.
+	 */
+	public static OMElement createRestServiceArtifact(OMElement wadlElement, String wadlName, String version, String wadlPath) {
+		if(wadlElement == null) {
+			throw new IllegalArgumentException("WADL content cannot be null." );
+		}
+		OMElement data = factory.createOMElement(CommonConstants.SERVICE_ELEMENT_ROOT, namespace);
+		OMElement overview = factory.createOMElement(OVERVIEW, namespace);
+		OMElement provider = factory.createOMElement(PROVIDER, namespace);
+		OMElement name = factory.createOMElement(NAME, namespace);
+		OMElement context = factory.createOMElement(CONTEXT, namespace);
+		OMElement apiVersion = factory.createOMElement(VERSION, namespace);
+		OMElement endpoint = factory.createOMElement(ENDPOINT_URL, namespace);
+		OMElement transports = factory.createOMElement(TRANSPORTS, namespace);
+		OMElement wadl = factory.createOMElement(WADL, namespace);
+		List<OMElement> uriTemplates = null;
+
+		provider.setText(CarbonContext.getThreadLocalCarbonContext().getUsername());
+		String serviceName = wadlName.contains(".") ? wadlName.substring(0, wadlName.lastIndexOf(".")) : wadlName;
+		name.setText(serviceName);
+		context.setText("/"+serviceName);
+		apiVersion.setText(version);
+		wadl.setText(wadlPath);
+		OMNamespace wadlNamespace = wadlElement.getNamespace();
+		String wadlNamespaceURI = wadlNamespace.getNamespaceURI();
+		String wadlNamespacePrefix = wadlNamespace.getPrefix();
+		OMElement resourcesElement =
+				wadlElement.getFirstChildWithName(new QName(wadlNamespaceURI, "resources", wadlNamespacePrefix));
+		if(resourcesElement != null) {
+			String endpointUrl =
+					resourcesElement.getAttributeValue(new QName("base"));
+			endpoint.setText(endpointUrl);
+			if(endpointUrl != null) {
+				transports.setText(endpointUrl.substring(0, endpointUrl.indexOf("://")));
+			}
+			uriTemplates = createURITemplateFromWADL(resourcesElement);
+		} else {
+			log.warn("WADL does not contains any resource paths. ");
+		}
+
+		overview.addChild(provider);
+		overview.addChild(name);
+		overview.addChild(context);
+		overview.addChild(apiVersion);
+		overview.addChild(transports);
+		overview.addChild(wadl);
+		overview.addChild(endpoint);
+		data.addChild(overview);
+
+		if (uriTemplates != null) {
+			for (OMElement uriTemplate : uriTemplates) {
+				data.addChild(uriTemplate);
+			}
+		}
+
+		return data;
+	}
+
+	/**
+	 * Saves the REST Service registry artifact created from the imported swagger definition.
+	 *
+	 * @param requestContext        information about current request.
+	 * @param data                  service artifact metadata.
+	 * @throws RegistryException    If a failure occurs when adding the api to registry.
+	 */
+	public static String addServiceToRegistry(RequestContext requestContext, OMElement data) throws RegistryException {
```

Comment:
@thushara35 I have reduced the complexity of this method by moving a generic code to CommonUtil class and also moving a part of a method to another method in same class. #218 (8a34525824c0891700033092c16b9b7222d27da5)